### DISCUSSION TITLE
Legrand Cable outlet (064879)

### DISCUSSION THREAD

**Discussion by EXP-Carthage** [Koenkk/zigbee2mqtt#11100] [Devices]
Hello,

First of all, please excuse me for my bad English.
I am programming the integration of the Legrand Cable outlet ([064882](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3786)) in Zigbee2MQTT.

For now, here is what is functional:
* On/Off mode management
* Pilot wire mode management
* Measurement of the power consumed

The passage in pilot wire mode allows to control a radiator according to the "signal" sent, which can be :
* Comfort
* Comfort -1
* Comfort -2
* Eco
* Frost protection
* Off

After some research, I discovered that you have to send a series of values to the device, which should look like :
`fcf + manufCode(2,4) + manufCode(0,2) + sqn + cmd + mode`

My temporary code to generate this data set looks like this :
``` javascript
const cable_outlet_mode = {
  'comfort': 0x00,
  'comfort-1': 0x01,
  'comfort-2': 0x02,
  'eco': 0x03,
  'frost_protection': 0x04,
  'off': 0x05
};
const fcf = "15";
const manufCode = "1021";
const sqn = "xx";  // TO FIND
const cmd = "00";
const mode = "0" + cable_outlet_mode[value];

const data = [
  fcf,
  manufCode.substring(2, 4),
  manufCode.substring(0, 2),
  sqn,
  cmd,
  mode
];
```

The problem is this SQN, I don't see where it is possible to get it...

I also can't find any function in the code to send this series of values.
As you can see, for this device, you have to send the bytes in this order :
`data[0] > data[1] > data[2] > data[3] > data[4] > data[5]`.

To finish this integration (and propose it on the github once finalized), I will greatly need your help.

**Comment by strati64** [comment]
I confirm, the absence of management of the “pilot wire” is currently a problem for all user with this.
I will probably go back to nodon SIN-4-FP-21 :/
